### ISSUE TITLE
Incorrect report of `no-unused-capturing-group` for `/^\.{1,2}(\/.*)?$/`

### ISSUE THREAD

**Issue by JounQin** [ota-meshi/eslint-plugin-regexp#443]
**Information:** v8.18.0
- ESLint version:
- `eslint-plugin-regexp` version: 1.7.0

**Description**

```ts
const RELATIVE_PATH_PATTERN = /^\.{1,2}(\/.*)?$/
```

`()?` should not be reported?

**Comment by ota-meshi** [comment]
If you don't use a capturing group, I think you can use a non-capturing group instead.

```js
const RELATIVE_PATH_PATTERN = /^\.{1,2}(?:\/.*)?$/
```

**Comment by JounQin** [comment]
Seems my regexp skill is not good enough then. 🤣

Can this plugin provide `suggestions`? (Auto fix maybe incorrect and risky.)

**Comment by ota-meshi** [comment]
> Seems my regexp skill is not good enough then. 🤣

I made this plugin and `no-unused-capturing-group` rule, because I made many similar mistakes before 😏

`no-unused-capturing-group` rule already supports the suggestion API.

https://github.com/ota-meshi/eslint-plugin-regexp/blob/4c72bde027cbfac45db233f3e3f96acca78e40c7/lib/rules/no-unused-capturing-group.ts#L86

If you tolerate the risk, you can optionally make an auto-fix.
https://ota-meshi.github.io/eslint-plugin-regexp/rules/no-unused-capturing-group.html#options

**Comment by JounQin** [comment]
closing since expected
